apps/constellation/settings.hpp

```diff
@@ -0,0 +1,120 @@
+#pragma once
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "core/feature_flags.hpp"
+#include "network/uri.hpp"
+#include "settings/setting.hpp"
+#include "settings/setting_collection.hpp"
+
+#include <cstdint>
+#include <string>
+#include <vector>
+
+namespace fetch {
+
+/**
+ * Command line / Environment variable settings
+ */
+class Settings : protected settings::SettingCollection
```

Comment:
inheritance may be private?

Reply:
Yep it could be in this case. `protected` seems a good idea in case we wanted to have derived configs in the future.

Reply:
Let's make it private for now, until something actually derives from it.